activesupport/lib/active_support/testing/assertions.rb

```diff
@@ -244,6 +244,22 @@ def assert_no_changes(expression, message = nil, from: UNTRACKED, &block)
 
         retval
       end
+
+      private
+        def _assert_nothing_raised_or_warn(&block)
+          assert_nothing_raised(&block)
+        rescue Minitest::UnexpectedError => e
+          if tagged_logger && tagged_logger.warn?
+            warning = <<~MSG
+              #{self.class} - #{name}: assert_nothing_raised called but #{e.error.class} raised.
```

Comment:
`assert_nothing_raised called` isn't ideal. In many cases you dont' call it yourself, you call e.g. `assert_difference {}`, which calls `assert_nothing_raised`. Maybe the caller name could be passed in argument or something.

Reply:
Yep, will make the warning more clear.

Reply:
I think the simplest way is just to remove the mention of `assert_nothing_raised` and keep the rest. So I did that.